packages/cli/mock/src/runMockServer.ts

```diff
@@ -0,0 +1,208 @@
+import { assertNever } from "@fern-api/core-utils";
+import { generateIntermediateRepresentation } from "@fern-api/ir-generator";
+import { TaskContext } from "@fern-api/task-context";
+import { FernWorkspace } from "@fern-api/workspace-loader";
+import {
+    ExampleEndpointCall,
+    ExamplePathParameter,
+    ExampleTypeReference,
+    HttpEndpoint,
+    NameAndWireValue
+} from "@fern-fern/ir-sdk/api";
+import express, { Request, Response } from "express";
+import { ParamsDictionary, Query } from "express-serve-static-core";
+import getPort from "get-port";
+import { IncomingHttpHeaders } from "http";
+import { isEqual } from "lodash-es";
+import qs from "qs";
+import urlJoin from "url-join";
+
+type RequestHandler = (req: Request, res: Response) => void;
+
+export async function runMockServer({
+    context,
+    workspace,
+    port
+}: {
+    context: TaskContext;
+    workspace: FernWorkspace;
+    port: number | undefined;
+}): Promise<void> {
+    const ir = await generateIntermediateRepresentation({
+        workspace,
+        audiences: { type: "all" },
+        generationLanguage: undefined
+    });
+
+    const app = express();
+    app.use(express.json({ limit: "50mb", strict: false }));
+
+    for (const service of Object.values(ir.services)) {
+        for (const endpoint of service.endpoints) {
+            const endpointPath = getFullPathForEndpoint(endpoint);
```

Comment:
i think we'll need to take account any top level api base path, service path and endpoint path for this to be accurate

Reply:
resolving due to use of `fullPath`

Reply:
We're actually using the full path here, so it will already include those details.